tests/res_diff_expect/fail_different_results_failures_2.ion

```diff
@@ -0,0 +1,25 @@
+$ion_1_0
+{
+ test: {
+  read_compare: {
+   message: "Read_compare: two revisions have different failures.",
+   failures: {
+    'ion-java_1': {
+     failures: ComparisonReport::[
+      {
+       message: "some errors 1",
+       lhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-java_1.ion"
+       },
+       rhs: {
+        location: "ion-test-driver/amazon/iontest/results/good/test.ion/read/data/ion-java_2.ion"
+       }
+      }
+     ]
+    },
+    'ion-java_2': {
```

Comment:
Doesn't like Errors, Failures should be equivalence between two revisions because they use its own cli tool to compare the same resources (e.g. read files generated by phase one). 

It's a little tricky between `same failures` and `same disagree list`, these are two different terms. we should check if the failures are same first to make sure cli tool performance same then we extract the disagree list from the failures and analyze the disagree list.

Reply:
I don't know if I fully understand your comment, but ion-java_1 had result `FAIL`, while ion-java_2 had result `PASS`. Shouldn't that always cause success since it is a move in the right direction?

Reply:
Sorry, I forgot to mentioned this earlier that PASS or FAIL may not be used for describing behavior good or not. it's just showing if we find difference between implementations.

For error field, it's showing the errors for an implementation so it's going good direction with less errors. For failures field, it's showing the difference between implementations so it's slightly different from error field that we always need to look into that and extract disagree list.

For example, the old implementation is able to find differences between two implementations so it shows 'FAIL' and writes failures into report while new one's compare command has issue and simply shows 'PASS'.

Reply:
Ah, I see. It's because the difference detected is between ion-java_1 and ion-java_2. So this must always be symmetrical, because they would be comparing the same data. Thanks for explaining.